docs/ProductionChecklistForScalarDBCluster.md

```diff
@@ -0,0 +1,140 @@
+
+You must always access ScalarDB Cluster. To ensure requests are running properly, check the properties file for your client application and confirm that `scalar.db.transaction_manager=cluster` is configured when you use CRUD API.
+
+* Recommended for production environments
+  ```mermaid
+  flowchart LR
+    A(<b>App</b> <br> ScalarDB Cluster Library with gRPC) --> B(<b>ScalarDB Cluster</b> <br> ScalarDB Library with Consensus Commit) --> C(Underlying storage/database)
+  ```
+
+* Not recommended for production environments (for testing purposes only)
+  ```mermaid
+  flowchart LR
+    A(<b>App</b> <br> ScalarDB Cluster Library with Consensus Commit) --> B(Underlying storage/database)
+  ```
+
+### SQL connection configuration (Java client library only)
+
+You must always access ScalarDB Cluster. To ensure requests are running properly, check the properties file for your client application and confirm that `scalar.db.sql.connection_mode=cluster` is configured when you use SQL API.
+
+* Recommended for production environments
+  ```mermaid
+  flowchart LR
+    A("<b>App</b> <br> ScalarDB SQL Library (Cluster mode)") --> B(<b>ScalarDB Cluster</b> <br> ScalarDB Library with Consensus Commit) --> C(Underlying storage/database)
+  ```
+
+* Not recommended for production environments (for testing purposes only)
+  ```mermaid
+  flowchart LR
+    A("<b>App</b> <br> ScalarDB SQL Library (Direct mode)") --> B(Underlying storage/database)
+  ```
+
+### Deployment of the client application when using `direct-kubernetes` client mode (Java client library only)
+
+If you use [`direct-kubernetes` client mode](https://github.com/scalar-labs/scalardb-cluster/blob/main/docs/developer-guide-for-scalardb-cluster-with-java-api.md#direct-kubernetes-client-mode), you must deploy your client application on the same Kubernetes cluster as the ScalarDB Cluster deployment.
+
+Also, when using `direct-kubernetes` client mode, you must deploy additional Kubernetes resources to make your client application work properly.  For details, see [Deploy your client application on Kubernetes with `direct-kubernetes` mode](https://github.com/scalar-labs/helm-charts/blob/main/docs/how-to-deploy-scalardb-cluster.md#deploy-your-client-application-on-kubernetes-with-direct-kubernetes-mode).
+
+### Transaction handling (Java client library and gRPC API)
+
+You must make sure that your application always runs [`commit()`](https://javadoc.io/static/com.scalar-labs/scalardb/3.10.0/com/scalar/db/api/DistributedTransaction.html#commit--) or [`rollback()`](https://javadoc.io/static/com.scalar-labs/scalardb/3.10.0/com/scalar/db/api/DistributedTransaction.html#rollback--) after you [`begin()`](https://javadoc.io/static/com.scalar-labs/scalardb/3.10.0/com/scalar/db/api/DistributedTransactionManager.html#begin--) a transaction. If the application does not run `commit()` or `rollback()`, your application might experience unexpected issues or read inconsistent data from the backend database.
```

Comment:
Should these links point to the Javadocs for ScalarDB 3.10.1 instead of 3.10.0?

(The following is something we can address later.) I worry that, if users click the links in this sentence, they'll be shown docs for a specific version rather than the version they might be using. For example, if a user is using 3.5, they'll be shown the Javadoc for 3.10. That might not be an issue now, but in the future, if the content changes, it might be confusing.

Reply:
I agree with your concerns. I investigated the way to specify `latest` in the URL of Javadoc, but I was not able to find it...

So, instead of Javadoc, I use the [ScalarDB Java API Guide](https://github.com/scalar-labs/scalardb/blob/master/docs/api-guide.md).

I updated it at d6b6fdc27b055ed81139df24eab4c4a01666418d.